spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/RemoveJsonAttributesResponseBodyGatewayFilterFactory.java

```diff
@@ -72,14 +72,14 @@ public GatewayFilter apply(FieldListConfiguration config) {
 		RewriteFunction<String, String> rewriteFunction = (exchange, body) -> {
 			if (MediaType.APPLICATION_JSON.isCompatibleWith(exchange.getResponse().getHeaders().getContentType())) {
 				try {
-					JsonNode jsonBodyContent = mapper.readValue(body, JsonNode.class);
+					JsonNode jsonNode = mapper.readValue(body, JsonNode.class);
 
-					removeJsonAttribute(jsonBodyContent, config.getFieldList(), config.isDeleteRecursively());
+					removeJsonAttributes(jsonNode, config.getFieldList(), config.isDeleteRecursively());
 
-					body = mapper.writeValueAsString(jsonBodyContent);
+					body = mapper.writeValueAsString(jsonNode);
 				}
 				catch (JsonProcessingException e) {
-					throw new RuntimeException(e);
+					return Mono.error(new IllegalStateException("Failed to process JSON of response body.", e));
```

Comment:
1. The caller wraps the error with `Mono.error()` to preserve the reactive pipeline and allow proper error handling.

2. The exception type has been explicitly changed to `IllegalStateException`.